### PR TITLE
Breaking changes for 2.0

### LINKED ISSUE #57: Breaking changes for 2.0
Get started with incorporating the breaking changes for 2.0 from OpenSearch 2.0.

To do so:
1. Branch off a 1.x from main if not already done so that development for 2.0 can start on main.
2. Incorporate breaking changes as mentioned in https://github.com/opensearch-project/opensearch-clients/issues/12.
3. Add integ tests support to run against unreleased OpenSearch.

Example PR from java-client: https://github.com/opensearch-project/opensearch-java/pull/132

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
This PR adds the breaking changes that come with OpenSearch 2.0. Changes in this PR include:

1. Support to test against unreleased OpenSearch.
2. Breaking changes mentioned in https://github.com/opensearch-project/opensearch-clients/issues/12
3. Documentation for upgrade. 
 
### Issues Resolved
Closes #57 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/test-unreleased.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Integration with Unreleased OpenSearch
```
Comment (saratvemulapalli) [inline .github/workflows/test-unreleased.yml:1]: May be add the workflow in a different PR?

Comment (VachaShah) [reply .github/workflows/test-unreleased.yml:1]: I added it in the same PR since the breaking changes need to be tested. WDYT? I can add the workflow in a separate PR and include only 1.x branch on it, then in this PR I can add the 2.x, 2.0 and main branch testing.

File: .github/workflows/test-unreleased.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    branches:
+      - "main"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        opensearch_ref: ['1.x', '2.x', '2.0', 'main']
```
Comment (dblock) [inline .github/workflows/test-unreleased.yml:17]: Nit: put these on separate lines so diffs are easier when adding/removing versions

**Review by dblock — APPROVED** [review]

File: .github/workflows/test-unreleased.yml (hunk 1 above)
Comment (saratvemulapalli) [reply .github/workflows/test-unreleased.yml:1]: Haha its ok. 
Its easier to split pieces when they are independent, we can do it next time :)

**Review by saratvemulapalli — APPROVED** [review]

**Review by dblock — APPROVED** [review]
